File: php/features.php
<?php
declare(strict_types=1);

// OsuBeatmap SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class OsuBeatmapFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new OsuBeatmapBaseFeature();
            case "test":
                return new OsuBeatmapTestFeature();
            default:
                return new OsuBeatmapBaseFeature();
        }
    }
}

File: py/features.py
# OsuBeatmap SDK feature factory

from feature.base_feature import OsuBeatmapBaseFeature
from feature.test_feature import OsuBeatmapTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OsuBeatmapBaseFeature(),
        "test": lambda: OsuBeatmapTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
